### PR TITLE
update Dockerfile, make the container behave similar to the real world

### PR DESCRIPTION
**Pull request by assafgi**

### CODE REVIEW

**Comment by assafgi** [comment]
https://pythonspeed.com/articles/base-image-python-docker-images/

File: talker_agent/Dockerfile [hunk 1]
```diff
@@ -8,9 +8,18 @@ RUN mkdir -p /root/talker
...
 RUN chmod +x install_agent.sh && ./install_agent.sh
 
-CMD ["python", "talker.py"]
+COPY talker-service /etc/init.d/talker
+RUN chmod +x /etc/init.d/talker
+RUN ln -s /usr/local/bin/python /usr/bin/python
+RUN echo "LC_ALL=en_US.UTF-8" >> /etc/environment
+RUN echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen
+RUN echo "LANG=en_US.UTF-8" > /etc/locale.conf
+RUN locale-gen en_US.UTF-8
```
Comment (DoronCohen) [inline talker_agent/Dockerfile]: move this up in the Docker file to better use caching

Comment (DoronCohen) [reply talker_agent/Dockerfile]: you can also make this a single command with `&&`

File: talker_agent/Dockerfile [hunk 2]
```diff
@@ -8,9 +8,18 @@ RUN mkdir -p /root/talker
...
-COPY talker.py install_agent.sh ./
+COPY talker.py talker-service install_agent.sh ./
+RUN chmod +x talker.py
 
-RUN apk add --no-cache curl bash
+RUN apt-get update -y && apt-get install curl locales -y
 RUN chmod +x install_agent.sh && ./install_agent.sh
 
-CMD ["python", "talker.py"]
+COPY talker-service /etc/init.d/talker
```
Comment (DoronCohen) [inline talker_agent/Dockerfile]: you already copy this in line 11

File: talker_agent/Dockerfile [hunk 3]
```diff
@@ -8,9 +8,18 @@ RUN mkdir -p /root/talker
 WORKDIR /root/talker
 
 RUN echo {\"password\": null, \"port\": 6379, \"host\": \"redis\", \"host_id\": \"$TALKER_HOST_ID\"} > conf.json
-COPY talker.py install_agent.sh ./
+COPY talker.py talker-service install_agent.sh ./
```
Comment (DoronCohen) [inline talker_agent/Dockerfile]: sort these steps in this order:
1. os config (installing dependencies, locale etc)
2. talker service copy and setup
3. talker.py copy and setup

File: talker_agent/Dockerfile [hunk 4]
```diff
@@ -7,10 +7,20 @@ ARG TALKER_HOST_ID
...
 
-RUN apk add --no-cache curl bash
+COPY install_agent.sh ./
 RUN chmod +x install_agent.sh && ./install_agent.sh
 
-CMD ["python", "talker.py"]
+COPY talker-service /etc/init.d/talker
+RUN chmod +x /etc/init.d/talker
+RUN ln -s /usr/local/bin/python /usr/bin/python
+RUN update-rc.d talker defaults
```
Comment (DoronCohen) [inline talker_agent/Dockerfile]: these can be combined to one `RUN` command

File: talker_agent/Dockerfile [hunk 5]
```diff
@@ -7,10 +7,20 @@ ARG TALKER_HOST_ID
 RUN mkdir -p /root/talker
 WORKDIR /root/talker
 
+RUN apt-get update -y && apt-get install curl locales -y
+RUN echo "LC_ALL=en_US.UTF-8" >> /etc/environment && echo "en_US.UTF-8 UTF-8" >> /etc/locale.gen && echo "LANG=en_US.UTF-8" > /etc/locale.conf
+RUN locale-gen en_US.UTF-8
```
Comment (DoronCohen) [inline talker_agent/Dockerfile]: these can be combined

**Pull request merged**